chore: v2.0.0-beta1 changelog (#1542)
google/osv-scanner: CHANGELOG.md

# v2.0.0-beta1

> This first beta of OSV-Scanner V2 is here! This beta release introduces significant enhancements, including refactored dependency extraction capabilities, container image scanning, and guided remediation for Maven.

Reviewer: ```suggestion
The first beta of OSV-Scanner V2 is here! This beta release introduces significant enhancements, including refactored dependency extraction capabilities, container image scanning, and guided remediation for Maven.
```

> This beta release does _not_ introduce any breaking CLI changes and the beta period is expected to last approximately one month. However, as this is a beta release, there may be breaking changes in the final release.

Reviewer 2: nit: breaking changes in the final release compared to the first beta? to be clear that we're not breaking v1 functionality.

# v2.0.0-beta1

> This beta release of OSV-Scanner introduces significant enhancements, including a refactored dependency extraction capabilities, container image scanning, and guided remediation for Maven. 

Reviewer: ```suggestion
This beta release of OSV-Scanner introduces significant enhancements, including refactored dependency extraction capabilities, container image scanning, and guided remediation for Maven. 
```

> This beta release does *not* introduce any breaking changes and the beta period is expected to last approximately one month. However, as this is a beta release, the API may be subject to any change.

Reviewer: Specify no breaking changes to the CLI, API is breaking.

We encourage you to try out these new features and provide feedback as we work towards the v2.0.0 final release.

### Enhanced Dependency Extraction with `osv-scalibr`

> The core dependency extraction logic has been rewritten to leverage the [`osv-scalibr`](https://github.com/google/osv-scalibr) library. We now support the following additional lockfile formats:

Reviewer: This also allows for easier/faster support for future formats

> The core dependency extraction logic has been rewritten to leverage the [`osv-scalibr`](https://github.com/google/osv-scalibr) library. We now support the following additional lockfile formats:

Reviewer 2: and also artifact formats? 

Also mention that more will come in future beta releases and with the final 2.0.0 release.

    * **Haskell:** `cabal.project.freeze`, `stack.yaml.lock`
    * **.NET:** `deps.json`
    * **Python:** `uv.lock`

### Container Image Scanning

> This release introduces the ability to scan container images for vulnerabilities, including support for Alpine and Debian images.

Reviewer: List out the supported OSes and languages:

Maybe with this table
```
| Distro Support | Language Artifacts Support |
| -------------- | -------------------------- |
| Alpine OS      | Go                         |
| Debian         | Java                       |
| Ubuntu         | Node                       |
|                | Python                     |
```

Reviewer: I think lets stick with node, mainly because it's common to have other javascript runtimes we don't support.

**Interactive HTML Reports:**  Vulnerability scan results for container images are presented in a new interactive HTML format, allowing for easy navigation and analysis of findings.

> 

Reviewer 3: Can we add an image of the HTML report here? link is https://github.com/user-attachments/assets/9e3e9c59-1948-45ab-9717-61fcbe3c7cc3

Reviewer 4: You could also link directly to `docs/images/html-container-output.png` (but that's a different image?)

### New Subcommands

Two new subcommands have been added for more targeted scanning:

* `scan source`:  Specifically for scanning manifest files.
* `scan image`:  Specifically for scanning container images.

### Guided Remediation for Maven

> OSV-Scanner can now automatically suggest and apply updates to vulnerable dependencies in your `pom.xml` file, which includes updates on both direct and transitive dependencies. We also support Maven private registry to find available pacakge verfsions.

Reviewer 4: typo:
```suggestion
OSV-Scanner can now automatically suggest and apply updates to vulnerable dependencies in your `pom.xml` file, which includes updates on both direct and transitive dependencies. We also support Maven private registry to find available package versions.
```

> OSV-Scanner can now automatically suggest and apply updates to vulnerable dependencies in your `pom.xml` file, which includes updates on both direct and transitive dependencies. We also support Maven private registry to find available pacakge verfsions.

Reviewer 4: Do we want a link to the guided remediation docs page for this?

Author: yes!

> OSV-Scanner can now automatically suggest and apply updates to vulnerable dependencies in your `pom.xml` file, which includes updates on both direct and transitive dependencies. We also support Maven private registry to find available pacakge verfsions.

Reviewer 2: link to our past blog post for npm and reference that
